pymeasure/instruments/novanta/fpu60.py

```diff
@@ -0,0 +1,141 @@
+
+    interlock_enabled = Instrument.measurement(
+        "INTERLOCK?",
+        """Measure the interlock enabled status.""",
+        values={True: "ENABLED", False: "DISABLED"},
+        map_values=True,
+    )
+
+    emission_enabled = Instrument.measurement(
+        "STATUS?",
+        """Measure the emission status. (bool)""",
+        values={True: "ENABLED", False: "DISABLED"},
+        map_values=True,
+    )
+
+    power = Instrument.measurement(
+        "POWER?",
+        """Measure current output power in Watts. (float)""",
+        # Response is in form:" ##.###W"
+        preprocess_reply=lambda r: r.replace("W", ""),
+    )
+
+    power_setpoint = Instrument.control(
+        "SETPOWER?", "POWER=%.3f",
+        """Control the output power setpoint in Watts. (float)""",
+        # Getter response is in form:" ##.###W"
+        preprocess_reply=lambda r: r.replace("W", ""),
+        check_set_errors=True,
+    )
+
+    shutter_open = Instrument.control(
+        "SHUTTER?", "SHUTTER %s",
+        """Control whether the shutter is open. (bool)""",
+        # set values: OPEN, CLOSE
+        # get response: "SHUTTER OPEN", "SHUTTER CLOSED"
+        values={True: "OPEN", False: "CLOSE"},
+        map_values=True,
+        preprocess_reply=lambda r: r.replace("SHUTTER ", "").replace("D", ""),
+        check_set_errors=True,
+    )
+
+    current = Instrument.measurement(
+        "CURRENT?",
+        """Measure the diode current in percent.""",
+        # Response: " ###.#%"
+        preprocess_reply=lambda r: r.replace("%", ""),
+    )
+
+    psu_temperature = Instrument.measurement(
+        "PSUTEMP?",
+        """Measure the power supply unit temperature in °C.""",
+        # Response: " ##.###C"
+        preprocess_reply=lambda r: r.replace("C", ""),
+    )
+
+    head_temperature = Instrument.measurement(
+        "HTEMP?",
+        """Measure the laser head temperature in °C.""",
+        # Response: " ##.###C"
+        preprocess_reply=lambda r: r.replace("C", ""),
+    )
+
+    serial = Instrument.measurement("SERIAL?", """Get the serial number""", cast=str)
+
+    software_version = Instrument.measurement("SOFTVER?", """Get the software version.""", cast=str)
+
+    def get_operation_times(self):
+        """Get the operation times in minutes as a dictionary."""
+        self.write("TIMERS?")
+        timers = {}
+        timers['psu'] = int(re.search(r"\d+", self.read()).group())
+        timers['laser'] = int(re.search(r"\d+", self.read()).group())
+        timers['laser_above_1A'] = int(re.search(r"\d+", self.read()).group())
+        self.read()  # an empty line is at the end.
+        return timers
+
+    def disable_emission(self):
+        """Disable emission and unlock the button afterwards.
+
+        You have to press the physical button to enable emission again.
+        """
+        self.ask("LASER=OFF")
+        self.ask("LASER=ON")  # unlocks emission button, does NOT start emission!
```

Comment:
Why do you use `self.ask` here, rather than `self.write`? Does the laser always respond after this command?

Reply:
Yes, the laser responds to every command.